api_request.go

```diff
@@ -39,40 +38,52 @@ func newDeadlineTransport(timeout time.Duration) *http.Transport {
 	return transport
 }
 
-func apiRequestNegotiateV1(method string, endpoint string, body io.Reader) (*simplejson.Json, error) {
+type generalResp struct {
+	Status     string      `json:"status_txt"`
+	StatusCode int         `json:"statuc_code"`
+	Data       interface{} `json:"data"`
+}
+
+func apiRequestNegotiateV1(method string, endpoint string, body io.Reader, ret interface{}) error {
```

Comment:
What did you mean "this never uses ret"?
`apiRequestNegotiateV1` may not know the type of response, so it couldn't return the data directly.